### ISSUE TITLE
Exporting string columns to pandas gives NaNs for NULL values

### ISSUE THREAD

**Issue by mvashishtha** [duckdb/duckdb#8652]
### What happens?

Null values for `VARCHAR` columns in duckdb become `NaN` in pandas. They should become `None`, which is the default missing value for string data. e.g. 

```python
import pandas as pd

pd.DataFrame([['Amsterdam', 1], [None, None]])
```

### To Reproduce

```python
import duckdb

duckdb.sql("SELECT * FROM (VALUES ('Amsterdam', 1), (NULL, NULL))").df()
```

### OS:

macOS Ventura 13.4 

### DuckDB Version:

0.8.1

### DuckDB Client:

Python

### Full Name:

Mahesh Vashishtha

### Affiliation:

Ponder Data

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by hannes** [comment]
Indeed this is expected behaviour

**Comment by mvashishtha** [comment]
> Our "NullValue" for VARCHAR is nullptr, we create a masked numpy array and that is later converted to Pandas.

@Tishj thank you for the explanation!

> Indeed this is expected behaviour

@hannes I disagree, and I suggest that we reopen this issue. I am a duckdb user and I did not expect this. I have used the pandas integrations of a few different database connectors in python and have so far only had this problem with duckdb. None of snowflake-connector-python, pyscopg3, the google bigquery connector, the pandas API on spark, or the redshift python connector, for example, have had this bug.

The behavior might be expected if you know that duckdb is trying to create pandas dataframes out of masked numpy arrays, and that pandas turns missing values in masked numpy arrays into `NaN`, but that doesn't make it desirable.

To start with addressing this issue, I suggest:

1) Open an issue with pandas about the behavior that we think may be a bug. I have opened for that. I have filed for https://github.com/pandas-dev/pandas/issues/54706
2) Meanwhile, see how much work it would take to work around 1), which may turn out to be expected behavior in pandas
